src/main/scala/mesosphere/marathon/api/GroupApiService.scala

```diff
@@ -33,11 +33,9 @@ class GroupApiService(groupManager: GroupManager)(implicit authorizer: Authorize
       case Some(version) =>
         val targetVersion = Timestamp(version)
         groupManager.group(group.id, targetVersion)
+          .map(_.getOrElse(throw new IllegalArgumentException(s"Group ${group.id} not available in version $targetVersion")))
```

Comment:
What motivated this change?

Reply:
this is consistent with the previously expected code but that code was incorrect.
AuthorizedAction[R] for `checkAuthorizationOrThrow[Resource](action: AuthorizedAction[Resource], resource: Resource)` is `Resource`.   `Resource` is expected to be of type `Group` but in this case prior to the change it was of type `Option[Group]`.   Once Auth was enforced properly it was required to map to the type from the option.